python/lsst/daf/butler/core/config.py

```diff
@@ -204,6 +205,8 @@ def __init__(self, other=None):
 
         if other is None:
             return
+        if isinstance(other, Path):
```

Comment:
I was imagining you'd add the instance check lower down where we check str, ButlerURI and then handoff to ButlerURI to deal with Path directly. Also need to change the docstring above to add `Path` along with `str` and `ButlerURI`.